src/teaching-element/Assessment/DragDrop.vue

```diff
@@ -126,6 +128,25 @@ export default {
       this.userAnswer[id] = this.userAnswer[id].filter(a => a.id !== answer.id);
       this.answersCollection.push(answer);
     },
+    retakeIncorrect() {
+      Object.keys(this.userAnswer).forEach(groupId => {
+        let wrongAnswers = [];
+        this.userAnswer[groupId].forEach((answer) => {
+          if (!includes(this.correct[groupId], answer.id)) {
+            wrongAnswers.push(answer.id);
+            this.answersCollection.push(answer);
+          }
+        });
+        this.removeFromGroup(wrongAnswers, groupId);
+      });
+    },
+    removeFromGroup(wrongAnswers, groupId) {
+      if (!wrongAnswers.length) return;
+      wrongAnswers.forEach(answer => {
+        const index = this.userAnswer[groupId].map(e => e.id).indexOf(answer);
```

Comment:
There is a [`findIndex`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) method on an array.